[lldb] Add support for PC-less scripted frames

medismailben opened the pull request:
Scripted frames that materialize Python functions or other non-native code are PC-less by design, meaning they don't have valid program counter values. Previously, these frames would display invalid addresses (`0xffffffffffffffff`) in backtrace output.

This patch updates `FormatEntity` to detect and suppress invalid address display for PC-less frames, adds fallback to frame methods when symbol context is unavailable, and modifies `StackFrame::GetSymbolContext` to skip PC-based symbol resolution for invalid addresses.

The changes enable PC-less frames to display cleanly with proper function names, file paths, and line numbers, and allow for source display of foreign sources (like Python). Includes comprehensive test coverage demonstrating frames pointing to Python source files.

jimingham commented on lldb/source/Core/FormatEntity.cpp:
```diff
@@ -1684,10 +1684,11 @@ bool FormatEntity::Format(const Entry &entry, Stream &s,
             return true;
-        }
+        } else if (frame->IsSynthetic())
+          return true;
```
Is returning true here right?  That would mean if I had `{pc: ${frame.pc}}` in my frame format, for a synthetic frame without a PC, this would still print the "pc: " part, which isn't correct.

medismailben commented on lldb/source/Core/CoreProperties.td:
```diff
@@ -57,10 +57,20 @@ let Definition = "debugger" in {
+            "[opt]}{${function.is-inlined} [inlined]}{${frame.is-artificial} "
+            "[artificial]}\\\\n">,
+        Desc<"The default frame format string to use when displaying stack "
+             "frame information for threads.">;
```
They are some changes in the `frame-format` & `frame-format-unique` strings. `clang-format` I had to reformat it because `clang-format` started complaining, I'll try to disable the formatting for this to make the diff more readable.

jimingham replied:
Why are we clang-formatting .td files anyway?  They aren't c source.

jimingham reviewed (approved):
LGTM

medismailben committed:
[lldb] Add support for PC-less scripted frames

Scripted frames that materialize Python functions or other non-native
code are PC-less by design, meaning they don't have valid program counter
values. Previously, these frames would display invalid addresses
(0xffffffffffffffff) in backtrace output.

This patch updates FormatEntity to detect and suppress invalid address
display for PC-less frames, adds fallback to frame methods when symbol
context is unavailable, and modifies StackFrame::GetSymbolContext to
skip PC-based symbol resolution for invalid addresses.

The changes enable PC-less frames to display cleanly with proper
function names, file paths, and line numbers, and allow for source
display of foreign sources (like Python). Includes comprehensive
test coverage demonstrating frames pointing to Python source files.

Signed-off-by: Med Ismail Bennani <ismail@bennani.ma>
